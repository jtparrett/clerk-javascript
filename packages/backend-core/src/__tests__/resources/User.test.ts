import { User } from '../../api/resources';
import { ObjectType, UserJSON } from '../../api/resources/JSON';

const mockUserDTO: UserJSON = {
  id: 'user_27jbYRV0tbvGovC63JFqaAqZsMy',
  object: ObjectType.User,
  username: 'test001',
  first_name: 'Test',
  last_name: 'Test',
  gender: '',
  birthday: '',
  profile_image_url: 'https://www.gravatar.com/avatar?d=mp',
  primary_email_address_id: 'idn_27jbYTsvdRnRyc5pFqOeD6ZkZ81',
  primary_phone_number_id: 'idn_27jbYTB1izaVaORupvdREuDqqaZ',
  primary_web3_wallet_id: 'test_web3wallet',
  password_enabled: true,
  two_factor_enabled: false,
  email_addresses: [
    {
      id: 'idn_27jbYTsvdRnRyc5pFqOeD6ZkZ81',
      object: ObjectType.EmailAddress,
      email_address: 'test@example.com',
      verification: {
        id: 'test_verification',
        object: ObjectType.Email,
        status: 'verified',
        strategy: 'email',
        attempts: 0,
        external_verification_redirect_url: '',
        expire_at: 12345,
        nonce: '',
      },
      linked_to: [],
    },
  ],
  phone_numbers: [
    {
      id: 'idn_27jbYTB1izaVaORupvdREuDqqaZ',
      object: ObjectType.PhoneNumber,
      phone_number: '+15555555555',
      reserved_for_second_factor: false,
      default_second_factor: false,
      verification: {
        id: 'test_verification',
        object: ObjectType.PhoneNumber,
        status: 'verified',
        strategy: 'phone',
        attempts: 0,
        external_verification_redirect_url: '',
        expire_at: 12345,
        nonce: '',
      },
      linked_to: [],
    },
  ],
  web3_wallets: [],
  external_accounts: [],
  public_metadata: { gender: 'female' },
  private_metadata: { middleName: 'Test' },
  unsafe_metadata: { unsafe: 'metadata' },
  external_id: 'a-unique-id',
  last_sign_in_at: 12345,
  created_at: 12345,
  updated_at: 12345,
};

test('user defaults', function () {
  const user = User.fromJSON(mockUserDTO);
  expect(user).toMatchSnapshot();
});
