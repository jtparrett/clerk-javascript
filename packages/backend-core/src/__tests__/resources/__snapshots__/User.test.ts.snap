// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user defaults 1`] = `
User {
  "birthday": "",
  "createdAt": 12345,
  "emailAddresses": Array [
    EmailAddress {
      "emailAddress": "test@example.com",
      "id": "idn_27jbYTsvdRnRyc5pFqOeD6ZkZ81",
      "linkedTo": Array [],
      "verification": Verification {
        "attempts": 0,
        "expireAt": 12345,
        "externalVerificationRedirectURL": null,
        "nonce": "",
        "status": "verified",
        "strategy": "email",
      },
    },
  ],
  "externalAccounts": Array [],
  "externalId": "a-unique-id",
  "firstName": "Test",
  "gender": "",
  "id": "user_27jbYRV0tbvGovC63JFqaAqZsMy",
  "lastName": "Test",
  "lastSignInAt": 12345,
  "passwordEnabled": true,
  "phoneNumbers": Array [
    PhoneNumber {
      "id": "idn_27jbYTB1izaVaORupvdREuDqqaZ",
      "linkedTo": Array [],
      "phoneNumber": "+15555555555",
      "reservedForSecondFactor": false,
      "verification": Verification {
        "attempts": 0,
        "expireAt": 12345,
        "externalVerificationRedirectURL": null,
        "nonce": "",
        "status": "verified",
        "strategy": "phone",
      },
    },
  ],
  "primaryEmailAddressId": "idn_27jbYTsvdRnRyc5pFqOeD6ZkZ81",
  "primaryPhoneNumberId": "idn_27jbYTB1izaVaORupvdREuDqqaZ",
  "primaryWeb3WalletId": "test_web3wallet",
  "privateMetadata": Object {
    "middleName": "Test",
  },
  "profileImageUrl": "https://www.gravatar.com/avatar?d=mp",
  "publicMetadata": Object {
    "gender": "female",
  },
  "twoFactorEnabled": false,
  "unsafeMetadata": Object {
    "unsafe": "metadata",
  },
  "updatedAt": 12345,
  "username": "test001",
  "web3Wallets": Array [],
}
`;
